build/ubuntu/Dockerfile.prebuilder

```diff
@@ -0,0 +1,74 @@
+FROM ubuntu:20.04 as cryptobox-builder
+
+# compile cryptobox-c
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update && \
+    apt-get install -y cargo file libsodium-dev git pkg-config && \
+    cd /tmp && \
+    git clone https://github.com/wireapp/cryptobox-c.git && \
+    cd cryptobox-c && \
+    export SODIUM_USE_PKG_CONFIG=1 && \
+    cargo build --release
+
+FROM ubuntu:20.04
+
+# install cryptobox-c in the new container
+COPY --from=cryptobox-builder /tmp/cryptobox-c/target/release/libcryptobox.so /usr/lib/libcryptobox.so
+COPY --from=cryptobox-builder /tmp/cryptobox-c/src/cbox.h /usr/include/cbox.h
+
+# development packages required for wire-server Haskell services
+RUN export DEBIAN_FRONTEND=noninteractive && \
+    apt-get update \
+    && apt-get install -y \
+        ca-certificates \
+        build-essential \
+        clang \
+        debhelper \
+        dh-autoreconf \
+        libgeoip-dev \
+        libglib2.0-dev \
+        libicu-dev \
+        libleveldb1d \
+        libleveldb-dev \
+        libossp-uuid-dev \
+        libpcre3-dev \
+        libsnappy-dev \
+        libssl-dev \
+        libstatgrab-dev \
+        pkg-config \
+        tcl \
+        upx-ucl \
+        zlib1g-dev \
+        libbz2-dev \
+        liblzma-dev \
+        liblzma5 \
+        libsodium-dev \
+        libsodium23 \
+        libpq-dev \
+        libxml2-dev \
+        libxml2 \
+        curl \
+        wget \
+        git \
+        libffi-dev \
+        libffi7 \
+        libgmp-dev \
+        libgmp10 \
+        libncurses-dev \
+        libncurses5 \
+        libtinfo5 \
+        protobuf-compiler
+
+# get static version of Haskell Stack and use system ghc by default
+ARG STACK_VERSION=2.7.3
+RUN curl -sSfL https://github.com/commercialhaskell/stack/releases/download/v${STACK_VERSION}/stack-${STACK_VERSION}-linux-x86_64-static.tar.gz \
+    | tar --wildcards -C /usr/local/bin --strip-components=1 -xzvf - '*/stack' && chmod 755 /usr/local/bin/stack && \
+    stack config set system-ghc --global true
+
+ARG GHC_VERSION=8.10.7
+RUN curl --proto '=https' --tlsv1.2 -sSf https://get-ghcup.haskell.org \
+  | BOOTSTRAP_HASKELL_NONINTERACTIVE=1 BOOTSTRAP_HASKELL_GHC_VERSION=${GHC_VERSION} sh
```

Comment:
why don't you install stack via ghcup?  seems easier and more robust?  (probably because it wasn't done before?)

Reply:
Oh right, ghcup can install stack too!

Reply:
Alright, done this. Thanks for the idea!